evalml/pipelines/components/estimators/estimator.py

```diff
@@ -21,6 +21,8 @@ class Estimator(ComponentBase):
 
     To see some examples, check out the definitions of any Estimator component.
     """
+    # We can't use the inspect module to dynamically determine this because of issue 1582
+    predict_uses_y = False
```

Comment:
We need a way of knowing whether or not to pass `y` as an argument to `predict` or `predict_proba` since one (more in the future) estimator allows this and others don't. I think this is the lowest-touch way of doing this.

One alternative is to list `y` as an argument for all estimators. However, I think that would be confusing for users to list `y` if it's not used because they might think our estimators are "cheating" by looking at the target value. Maybe I'm just being paranoid though.